packages/rxfire/firestore/collection/index.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { firestore } from 'firebase/app';
+import { fromCollectionRef } from '../fromRef';
+import { Observable } from 'rxjs';
+import { map, filter, scan } from 'rxjs/operators';
+
+const ALL_EVENTS: firestore.DocumentChangeType[] = [
+  'added',
+  'modified',
+  'removed'
+];
+
+/**
+ * Create an operator that determines if a the stream of document changes
+ * are specified by the event filter. If the document change type is not
+ * in specified events array, it will not be emitted.
+ */
+const filterEvents = (events?: firestore.DocumentChangeType[]) =>
+  filter((changes: firestore.DocumentChange[]) => {
+    let hasChange = false;
+    changes.forEach(change => {
+      if (events.indexOf(change.type) >= 0) {
+        hasChange = true;
+      }
+    });
+    return hasChange;
+  });
+
+/**
+ * Create an operator that filters out empty changes. We provide the
+ * ability to filter on events, which means all changes can be filtered out.
+ * This creates an empty array and would be incorrect to emit.
+ */
+const filterEmpty = filter(
+  (changes: firestore.DocumentChange[]) => changes.length > 0
+);
+
+/**
+ * Creates a new sorted array from a new change.
+ * @param combined
+ * @param change
+ */
+function combineChange<T>(
+  combined: firestore.DocumentChange[],
```

Comment:
Let's be more descriptive of what this actually is in this name (i.e. the full array of changes and a single change to process)

Reply:
fixed.